src/commands/scheduledQuestion.ts

```diff
@@ -0,0 +1,146 @@
+          { name: 'NHIE', value: 'NHIE' },
+          { name: 'PARANOIA', value: 'PARANOIA' },
+          { name: 'random', value: 'NONE' },
+        ],
+      },
+      {
+        type: ApplicationCommandOptionType.String,
+        name: 'rating',
+        description: 'The maturity level of the topics the question can relate to.',
+        choices: [
+          { name: 'PG', value: 'PG' },
+          { name: 'PG13', value: 'PG13' },
+          { name: 'R', value: 'R' },
+          { name: 'random', value: 'NONE' },
+        ],
+      },
+      {
+        type: ApplicationCommandOptionType.Role,
+        name: 'role',
+        description: 'The role to ping whenever a question is sent.',
+      },
+    ],
+  },
+  {
+    type: ApplicationCommandOptionType.Subcommand,
+    name: 'remove',
+    description: 'Remove a scheduled question set in the channel.',
+  },
+  {
+    type: ApplicationCommandOptionType.Subcommand,
+    name: 'view',
+    description: 'See all scheduled question channels you have set up.',
+  },
+] as const;
+
+const scheduledQuestion: Command = {
+  name: 'scheduledquestion',
+  description: 'Automatically post a question every certain amount of time!',
+  options,
+  category: 'control',
+  perms: ['ManageGuild'],
+  run: async (ctx: Context) => {
+    if (!ctx.guildId)
+      return ctx.reply(ctx.client.EMOTES.xmark + ' This command cannot be run in DMs.');
+
+    if (!ctx.premium) return ctx.replyUpsell();
+
+    if (ctx.args[0] === 'create') {
+      if (await ctx.client.database.getScheduledQuestionChannel(ctx.channelId))
+        return ctx.reply(
+          `${ctx.client.EMOTES.xmark} There's already a scheduled question in this channel. Please remove the old one with \`/scheduledquestion remove\` before creating a new one!`
+        );
+
+      const frequency = ctx.getOption<Mutable<typeof options[0]['options'][0]>>('frequency')!.value;
+      let questionType =
+        ctx.getOption<Mutable<typeof options[0]['options'][1]>>('type')?.value || null;
+      let rating = ctx.getOption<Mutable<typeof options[0]['options'][2]>>('rating')?.value || null;
+      const pingRole =
+        ctx.getOption<Mutable<typeof options[0]['options'][3]>>('role')?.value || null;
+
+      if (questionType === 'NONE') questionType = null;
+      if (rating === 'NONE') rating = null;
+
+      await ctx.client.database.createScheduledQuestionChannel({
+        id: ctx.channelId,
+        botId: ctx.client.id,
+        guildId: ctx.guildId,
+        schedule: frequency,
+        type: questionType,
+        rating,
+        role: pingRole,
+      });
+
+      return ctx.reply(
+        `${ctx.client.EMOTES.checkmark} Created a ${frequency.toLowerCase()} scheduled question!`
+      );
+    } else if (ctx.args[0] === 'remove') {
+      await ctx.client.database.deleteScheduledQuestionChannel(ctx.channelId);
+
+      return ctx.reply(
+        `${ctx.client.EMOTES.checkmark} Removed the scheduled question in this channel.`
+      );
```

Comment:
This doesn't check if a scheduled question exists, so you can run it without anything configured and the bot responds with success.


<img width="410" alt="CleanShot 2022-12-16 at 13 19 54@2x" src="https://user-images.githubusercontent.com/36977340/208163557-9a65178d-7d6b-441b-8a54-208e6939f556.png">

Reply:
lol